components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -155,6 +228,88 @@ ConversationHandler* AIChatService::CreateConversationHandlerForContent(
   return conversation;
 }
 
+void AIChatService::ClearAllHistory() {
+  // Delete opt-in state
+  profile_prefs_->ClearPref(prefs::kLastAcceptedDisclaimer);
+
+  // Delete in-memory data
+  conversation_observations_.RemoveAllObservations();
+  conversation_handlers_.clear();
+  conversations_.clear();
+  content_conversations_.clear();
+  OnConversationListChanged();
+
+  // Delete database data
+  if (ai_chat_db_) {
+    ai_chat_db_.AsyncCall(&AIChatDatabase::DeleteAllData)
+        .Then(base::BindOnce([](bool success) {
+          if (!success) {
+            LOG(ERROR) << "Failed to delete all AIChatService database data";
+          }
+        }));
+  }
+  if (ai_chat_metrics_ != nullptr) {
+    ai_chat_metrics_->RecordReset();
+  }
+}
+
+void AIChatService::OnOsCryptAsyncReady(const base::FilePath& storage_dir,
+                                        os_crypt_async::Encryptor encryptor,
+                                        bool success) {
+  CHECK(features::IsAIChatHistoryEnabled());
+  if (!success) {
+    LOG(ERROR) << "Failed to initialize AIChat DB due to OSCrypt failure";
+    return;
+  }
+
+  ai_chat_db_ = base::SequenceBound<AIChatDatabase>(
+      GetDBTaskRunner(), storage_dir.Append(kDBFileName), std::move(encryptor));
+
+  ai_chat_db_.AsyncCall(&AIChatDatabase::IsInitialized)
+      .Then(base::BindOnce(&AIChatService::OnDBInit,
+                           weak_ptr_factory_.GetWeakPtr()));
+}
+
+void AIChatService::OnDBInit(bool success) {
+  CHECK(ai_chat_db_);
+  if (!success) {
+    LOG(ERROR) << "Failed to initialize AIChat DB";
+    // Ensure any pending tasks are cancelled. Sometimes, especially during unit
+    // tests that are very quick to run, initilization might fail because of
+    // shutdown but the next task in the sequence still executes, causing a
+    // crash because of bad database state.
+    ai_chat_db_.Reset();
+    return;
+  }
```

Comment:
I did start out by having a separate public AIChatDatabase::Init function that consumers would call, but it seemed redundant because what's the point of constructing the class without Init being run, and also I didn't want to have to check that it was init'ed on every other function call.

I could change it back, but just thinking about the reasoning - is it really possible for `is_initialized_` to be checked in the middle of the initializing process when all the other functions will be called on the sequence and I assume that guarantees Init will be finished, since it's called in the constructor? I know I'm probably wrong about this so please let me know!

Reply:
But even with current code, we still rely on consumer to check if it is successfully inited and it is not safe to assume that it would successfully initialized upon calling other public functions. So we still miss checking `is_initialized_` in all public functions because init might fail.

Reply:
I've changed this to Init internally on every public function, and to do it lazily. This mirrors a pattern from recent Chromium code, e.g. `components/history_embeddings/sql_database.cc`. See `LazyInit`.